src/json_parser.cc

```diff
@@ -34,16 +40,22 @@ bool JSONParser::Parse(const std::string& content) {
       !result_value->IsObject()) {
     return false;
   }
+
+  context_.Reset(isolate, context);
   content_.Reset(isolate, result_value.As<Object>());
   parsed_ = true;
+
   return true;
 }
 
 std::optional<std::string> JSONParser::GetTopLevelStringField(
     std::string_view field) {
   Isolate* isolate = isolate_.get();
+  HandleAndContextScope handle_and_context_scope(isolate, context_);
```

Comment:
This helper class can be expanded to two lines:

```suggestion
  HandleScope handle_scope(isolate);
  Context::Scope context_scope(context_.Get(isolate));
```

As HandleScope and ContextScope are established V8 APIs, I'd find it to be superfluous to add a helper class for them.

Reply:
It'll be three lines soon, when I add Isolate::Scope, but I think you're still right that it's cleaner expanded out.